src/Raven.Server/Documents/Sharding/ShardingRecordExtensions.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using Raven.Client.ServerWide;
+using Raven.Client.ServerWide.Sharding;
+using Sparrow.Utils;
+using Voron;
+
+namespace Raven.Server.Documents.Sharding;
+
+public static class ShardingRecordExtensions
+{
+    public static List<ShardBucketRange> For(this ShardingRecord record, string id)
+    {
+        if (record.Prefixed == null)
+            return record.ShardBucketRanges;
+
+        foreach (var kvp in record.Prefixed)
+        {
+            if (id.StartsWith(kvp.Key, StringComparison.OrdinalIgnoreCase))
```

Comment:
is it legal to have prefixes like `foo` & `foobar` ?
what happened if we modify this? I guess the new prefix configuration will be valid only for new documents? what about modifying a document? then we need to move it with all his revisions/TS etc?

Reply:
Actually if modified we must move all relevant documents, otherwise basic things like `Load` wouldn't work

Reply:
But this a problem for any `ShardBucketRange` modification..

Reply:
They are handled in order, and we'll probably insist on `/` postfix to ensure it works.
But in general, not a problem.